perf/sawtooth_workload/src/main.rs

```diff
@@ -120,9 +120,7 @@ fn run_batch_command(args: &ArgMatches) -> Result<(), Box<Error>> {
     try!(key_file.read_to_string(&mut buf));
     buf.pop(); // remove the new line
 
-    let private_key = try!(
-        Secp256k1PrivateKey::from_hex(&buf).or(
-            Secp256k1PrivateKey::from_wif(&buf)));
+    let private_key = try!(Secp256k1PrivateKey::from_wif(&buf)).or(Secp256k1PrivateKey::from_hex(&buf))
```

Comment:
Good catch, but wrong fix - it should be:
```
let private_key = try!(Secp256k1PrivateKey::from_hex(&buf)).or_else(|_| Secp256k1PrivateKey::from_wif(&buf))
```

Reply:
Not sure what the difference between the `or` and `or_else` syntax (I'm kindof new to Rust) but I trust your judgement here and changed to use `or_else` and changed the order back to `from_hex` followed by `from_wif`. I guess it is theoretically possible to have a wif format key that has a base58 encoding that is actually only using the 0-9a-f characters, but that's probably so rare that we don't need to worry about it here!

Reply:
Oh, right.  Hadn't thought of that - so probably both reordering _and_ using `or_else` is the solution.

As for the why `or_else` over `or`?  In the `or_else` case, the code is only executed on a failure.  In the `or` case, both are executed, and the successful one is chosen.  I.e. less work in the former over the later.

Reply:
OK, I understand the `or_else` rationale now, thanks. Also, you were right with the ordering, so I changed it back to `from_hex` then `from_wif`. Once I thought about it a bit more, you _need_ to try hex first, because any valid hex is _also_ valid base58 (so most wif keys will fail to parse as hex, but hex keys will parse as wif) and I believe the wif keys have a different length anyway, so they should _still_ fail to parse even if they have pathological formatting of only `0-9a-f` characters!